fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/Lambda.java

```diff
@@ -0,0 +1,162 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.nereids.trees.expressions.functions.scalar;
+
+import org.apache.doris.nereids.trees.expressions.ArrayItemReference;
+import org.apache.doris.nereids.trees.expressions.Expression;
+import org.apache.doris.nereids.trees.expressions.visitor.ExpressionVisitor;
+import org.apache.doris.nereids.types.ArrayType;
+import org.apache.doris.nereids.types.DataType;
+import org.apache.doris.nereids.types.LambdaType;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableList.Builder;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+/**
+ * Lambda includes lambda arguments and function body
+ * Before bind, x -> x : arguments("x") -> children: Expression(x)
+ * After bind, x -> x : arguments("x") -> children: Expression(x) ArrayItemReference(x)
+ */
+public class Lambda extends Expression {
+    final List<String> argumentNames;
+
+    /**
+     * constructor
+     */
+    public Lambda(List<String> argumentNames, Expression lambdaFunction) {
+        super(lambdaFunction);
+        this.argumentNames = argumentNames;
+    }
+
+    public Lambda(List<String> argumentNames, Expression lambdaFunction, List<ArrayItemReference> arguments) {
+        super(ImmutableList.<Expression>builder().add(lambdaFunction).addAll(arguments).build());
+        this.argumentNames = argumentNames;
+    }
+
+    public Lambda(List<String> argumentNames, List<Expression> children) {
+        super(children);
+        this.argumentNames = argumentNames;
+    }
+
+    /**
+     * make slot according array expression
+     * @param arrays array expression
+     * @return item slots of array expression
+     */
+    public ImmutableList<ArrayItemReference> makeArguments(List<Expression> arrays) {
+        Builder<ArrayItemReference> builder = new ImmutableList.Builder<>();
+        for (int i = 0; i < arrays.size(); i++) {
+            Expression array = arrays.get(i);
+            String name = argumentNames.get(i);
+            Preconditions.checkArgument(array.getDataType() instanceof ArrayType, "lambda must receive array");
```

Comment:
throw new AnalysisException(). and print the original expression for user

Reply:
done